build/openresty/patches/ngx_lua-0.10.25_01-dyn_upstream_keepalive.patch

```diff
@@ -1,8 +1,33 @@
+diff --git a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
+index b07e564..68c9b05 100644
+--- a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
++++ b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.c
+@@ -4304,6 +4304,7 @@ ngx_http_upstream_next(ngx_http_request_t *r, ngx_http_upstream_t *u,
+     if (u->peer.cached && ft_type == NGX_HTTP_UPSTREAM_FT_ERROR) {
+         /* TODO: inform balancer instead */
+         u->peer.tries++;
++        u->peer.notify(&u->peer, u->peer.data, NGX_HTTP_UPSTREAM_NOTIFY_BALANCER);
+     }
+ 
+     switch (ft_type) {
+diff --git a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.h b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.h
+index a385222..d3de3b7 100644
+--- a/bundle/nginx-1.21.4/src/http/ngx_http_upstream.h
++++ b/bundle/nginx-1.21.4/src/http/ngx_http_upstream.h
+@@ -56,6 +56,8 @@
+ #define NGX_HTTP_UPSTREAM_IGN_VARY           0x00000200
+ 
+ 
++#define NGX_HTTP_UPSTREAM_NOTIFY_BALANCER    0x1
```

Comment:
The readers don't seem to be able to tell from the name what type of notification it is. What about using a more specific name? e.g. NGX_HTTP_UPSTREAM_NOFITY_CACHED_PEER_ERROR

Reply:
It is not an error, when `NGX_HTTP_UPSTREAM_FT_ERROR` occurs and `cached = 1` ( use the connection used from the pool), it is necessary to notify the balancer to handle it on its own.

If there is no `ngx_lua_balancer` and dynamic upstream pool patch, the original code here cannot be considered an error.

so `NGX_HTTP_UPSTREAM_NOFITY_CACHED_PEER_ERROR` can be `NGX_HTTP_UPSTREAM_NOTIFY_BALANCER_UPSTREAM_FT_ERROR` but it too lang...

Reply:
rename to `NGX_HTTP_UPSTREAM_NOFITY_CACHED_CONNECTION_ERROR`